lib/rubocop/cop/lint/void.rb

```diff
@@ -217,6 +217,17 @@ def autocorrect_nonmutating_send(corrector, node, suggestion)
                       end
           corrector.replace(send_node.loc.selector, suggestion)
         end
+
+        def entirely_literal?(node)
+          case node.type
+          when :array
+            node.each_value.all?(&:literal?)
+          when :hash
+            node.each_value.all?(&:literal?) && node.each_key.all?(&:literal?)
```

Comment:
should this be recursive to handle nested literals like `[1, [2, 3]]`?

Reply:
Considering the possibility, it should have been addressed. I've updated this PR. Thank you!